src/visualizers/widgets/TextEditor/TextEditorWidget.js

```diff
@@ -1,79 +1,111 @@
-/*globals $, define*/
+/*globals $, define, monaco */
 /*jshint browser: true*/
 
 define([
-    'ace/ace',
     'underscore',
-    './completer',
     'js/Utils/ComponentSettings',
+    'text!./MonacoLanguages.json',
+    'vs/editor/editor.main',
     'jquery-contextMenu',
     'css!./styles/TextEditorWidget.css'
 ], function (
-    ace,
     _,
-    Completer,
-    ComponentSettings
+    ComponentSettings,
+    MonacoLanguages
 ) {
     'use strict';
 
-    var TextEditorWidget,
-        WIDGET_CLASS = 'text-editor',
+    const WIDGET_CLASS = 'text-editor',
         LINE_COMMENT = {
             python: '#',
             lua: '--',
             yaml: '#'
         };
 
-    TextEditorWidget = function (logger, container, config={}) {
-        this.logger = logger.fork('Widget');
+    MonacoLanguages = JSON.parse(MonacoLanguages);
+    const DEFAULT_THEMES = ['vs-dark', 'vs', 'hc-black'];
+    const DEFAULT_COLORS = {
```

Comment:
It looks like this can be removed.

Reply:
Due to the lack of submenu support and already exsiting context menu. My plan was to add a separate button. This which opems the existing contextmenu.